File: frontend/src/app/core/services/cliente.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { ApiResponse, PagedResponse } from '../models/api-response.model';
import { Cliente } from '../models/cliente.model';

@Injectable({ providedIn: 'root' })
export class ClienteService {
  constructor(private api: ApiService) {}

  getAll(params?: any): Observable<ApiResponse<PagedResponse<Cliente>>> {
    return this.api.getPaged('/clientes', params?.page, params?.size, params);
  }

  getById(id: number): Observable<ApiResponse<Cliente>> {
    return this.api.get(`/clientes/${id}`);
  }

  create(data: Partial<Cliente>): Observable<ApiResponse<Cliente>> {
    return this.api.post('/clientes', data);
  }

  update(id: number, data: Partial<Cliente>): Observable<ApiResponse<Cliente>> {
    return this.api.put(`/clientes/${id}`, data);
  }

  delete(id: number): Observable<ApiResponse<void>> {
    return this.api.delete(`/clientes/${id}`);
  }
}

File: backend/src/main/java/bo/capital/tec/pet/service/ClienteService.java
package bo.capital.tec.pet.service;

import bo.capital.tec.pet.common.response.PagedResponse;
import bo.capital.tec.pet.dto.ClienteRequestDTO;
import bo.capital.tec.pet.dto.ClienteResponseDTO;
import bo.capital.tec.pet.dto.ClienteSummaryDTO;

public interface ClienteService {
    ClienteResponseDTO create(ClienteRequestDTO dto);
    ClienteResponseDTO getById(Long id);
    ClienteResponseDTO getByPersonaId(Long personaId);
    PagedResponse<ClienteSummaryDTO> getAll(String nombre, String ci, boolean activo, int page, int size);
    ClienteResponseDTO update(Long id, ClienteRequestDTO dto);
    void delete(Long id);
}
